lib/services/devices/registrationUtils.js

```diff
@@ -158,5 +201,148 @@ function sendRegistrations(unregister, deviceData, callback) {
     }
 }
 
+/**
+ * Sends a Context Provider unregistration request to the Context Broker using NGSIv2.
+ *
+ * @param {Object} deviceData       Object containing all the deviceData needed to send the registration.
+ */
+function sendUnregistrationsNgsi2(deviceData, callback) {
+    var options = {
+        url: config.getConfig().contextBroker.url + '/v2/registrations/' + deviceData.registrationId,
+        method: 'DELETE',
+        json: true,
+        headers: {
+            'fiware-service': deviceData.service,
+            'fiware-servicepath': deviceData.subservice
+        }
+    };
+
+    logger.debug(context, 'Sending device unregistrations to Context Broker at [%s]', options.url);
+    logger.debug(context, 'Using the following request:\n\n%s\n\n', JSON.stringify(options, null, 4));
+
+    request(options, createRegistrationHandlerNgsi2(true, deviceData, callback));
+}
+
+/**
+ * Sends a Context Provider registration or unregistration request to the Context Broker using NGSIv2.
+ *
+ * @param {Boolen} unregister       Indicates whether this registration is an unregistration or register.
+ * @param {Object} deviceData       Object containing all the deviceData needed to send the registration.
+ */
+function sendRegistrationsNgsi2(unregister, deviceData, callback) {
+    var options = {
+        url: config.getConfig().contextBroker.url + '/v2/registrations',
+        method: 'POST',
+        json: {
+            dataProvided: {
+                entities:
+                [
+                    {
+                        type: deviceData.type,
+                        id: String(deviceData.name)
+                    }
+                ],
+                attrs: [],
+            },
+            provider: {
+                http: {
+                    url: config.getConfig().providerUrl
+                },
+                legacyForwarding: true
+            }
+        },
+        headers: {
+            'fiware-service': deviceData.service,
+            'fiware-servicepath': deviceData.subservice
+        }
+    };
+
+    function formatAttributes(originalVector) {
+        var attributeList = [];
+
+        if (originalVector && originalVector.length) {
+            for (var i = 0; i < originalVector.length; i++) {
+                attributeList.push(originalVector[i].name);
+            }
+        }
+
+        return attributeList;
+    }
+
+
+    function mergeWithSameName(old, current) {
+        if (old.indexOf(current) < 0) {
+            old.push(current);
+        }
+
+        return old;
+    }
+
+    // FIXME: When https://github.com/telefonicaid/fiware-orion/issues/3007 is merged into master branch,
+    // this function should use the new API. This is just a temporary solution which implies deleting the
+    // registration and creating a new one.
+    function updateRegistrationNgsi2(deviceData, callback) {
+        var functions = [];
+
+        function removeRegistrationId(deviceData, otracosa, callback) {
```

Comment:
"otracosa" ? ;)

Reply:
Fixed with 327288a44634ccf510ae5aff4ca5955a1eaafef9

Reply:
I forgot to rename that parameter! :P